### DISCUSSION TITLE
N00b questions! Syncing, selecting a directory and sub-dirs to sync, ommitting dirs from sync

### DISCUSSION THREAD

**Discussion by gdg23** [abraunegg/onedrive#2718] [General]
Hello,

Newly installed, looks great but I haven't tried to sync anything yet. I'm just a little unclear on a few things:

- At the OneDrive 'top level' I have a few directories but I'm only interested in my work folder. This has a lot of sub-folders and sub-sub-...folders. I ran:

onedrive --dry-run --synchronize --single-directory 'my_work_dir'

and it looks like that was going to do the right thing (only include my work folder and all of its sub-folders). 

How would I change the config file to reflect this? I'm assuming that I should change the config file so that future syncs also only include this top-level dir (and sub dirs) (even if the CLI command does what I want for the _initial_ sync)? I see I can set 'sync_dir' in the config file but I'm not sure what that should look like. "~/OneDrive/my_work_dir" maybe? Is the value of sync_dir referring to a dir on my local machine or the OneDrive directory (or both)?

- I'd also like to exclude a couple of dirs from within my_work_dir (that also have sub-dirs), those dirs are a couple deep, like "my work dir/sub-dir/dir-to-exclude". I see there's a skip_dir option which looks like it does this, similar to the above my question is really just about syntax and the path to use in the config file. Would it be:

skip_dir = "/my_work_dir/sub_folder/folder_to_exclude"   ?

- And lastly, when I make changes to my files (once I sync them), are those changes automatically synced back to OneDrive?

Thanks, and apologies for the slightly confused questions!

Guy

**Comment by abraunegg** [comment]
@gdg23 
Firstly thanks for using this client.

To answer most of your questions, please read the documentation:
* https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md (older but still relevant)
* https://github.com/abraunegg/onedrive/blob/onedrive-v2.5.0-release-candidate-2/docs/usage.md (new doc version)
* https://github.com/abraunegg/onedrive/blob/onedrive-v2.5.0-release-candidate-2/docs/application-config-options.md

> How would I change the config file to reflect this?
> ...
> I see I can set 'sync_dir' in the config file but I'm not sure what that should look like. "~/OneDrive/my_work_dir" maybe?

Do not do this. Please read the documentation on what this option represents.

> And lastly, when I make changes to my files (once I sync them), are those changes automatically synced back to OneDrive?

To do this, you need to run a systemd service so that changes made locally are automatically uploaded. Again, please read the documentation on this.

**Reply by MMUD74** [reply]
Hello I Have a question, can I sync from my OMV to my onedrive? Or can it only sync from onedrive down to my OMV share?

😊 thank you

**Reply by abraunegg** [reply]
@MMUD74 
> Hello I Have a question, can I sync from my OMV to my onedrive?

Yes you can use 'onedrive' via 2 options:
* Install via the OMV plugin
* Install *normally* for distributions based on Debian & Ubuntu: https://github.com/abraunegg/onedrive/blob/master/docs/ubuntu-package-install.md

> Or can it only sync from onedrive down to my OMV share?

You can configure the client to perform a number of sync operations. 

Please *read* the documentation:
* https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md
* https://github.com/abraunegg/onedrive/blob/onedrive-v2.5.0-release-candidate-2/docs/usage.md
* https://github.com/abraunegg/onedrive/blob/onedrive-v2.5.0-release-candidate-2/docs/application-config-options.md


>[!IMPORTANT]
> If you choose to use the OMV provided package and OMV Plugin you are extremely limited in client functionality due to how OMV has integrated the client. To use the client *correctly* I recommend you manually install the client and configure the client to operate how you want.
